### DISCUSSION TITLE
List of Known Working Third Party UEs

### DISCUSSION THREAD

**Discussion by jtplatt99** [srsran/srsRAN_Project#36] [User Success Stories]
Hello All!

I thought it would be valuable for the user-community and developers to put together a list of known UE devices that have been tested with srsRAN and share any information about getting them to work.

So far, I've had success connecting the following UEs:
- OnePlus 9 Pro (LE2125)
- Quectel RM510Q-GL
- Amarisoft AMARI UE Simbox

I've been using a USRP N320 as my radio at n78 at 20 MHz bandwidth with 30 kHz subcarrier spacing. I'm using an Open5GS core. Here is my full configuration file: [gnb_rf_n320_tdd_n78_20mhz.zip](https://github.com/srsran/srsRAN_Project/files/11042320/gnb_rf_n320_tdd_n78_20mhz.zip).

One question for the community and developers: is anyone else using an Amarisoft AMARI UE Simbox? With our current configuration the UE Simbox is experiencing >90% bit-error rate and only a few kbps on iperf3 while our other UEs easily get >6 Mbps on iperf3. If anyone else out there is using an Amarisoft AMARI UE Simbox with srsRAN, I'm hoping to compare set ups and throughput numbers.

**Comment by gustavobsch** [comment]
I've had success connecting the following UEs:

OnePlus nord 10 BE2029

No sucess with 

Galaxy A22

**Reply by ismagom** [reply]
Hi @gustavobsch , It would be very helpful if you could provide us logs (info is enough) & pcaps for the Galaxy A22. 

Thanks

**Reply by gustavobsch** [reply]
I will test again with external GPSDO attached to the B210.. the problem with the Galaxy A22 is that I don't see the network at all.. whereas the Oneplus I can see the network even without a proper SIM inserted.

**Reply by ismagom** [reply]
Try combining SIM cards and PLMNs. For instance with the OnePlus Nord we've seen the following combination to work: 

- IMSI starting with 00101... and network PLMN 90170

Any other IMSI or PLMN combination doesn't seem to work

**Reply by ankuramistry** [reply]
Hi @ismagom @jtplatt99 , 

We are trying to connect Amari UE Simbox with the USRPN300,  srsRAN and open5GS and have been unsuccessful so far. There is no messaging taking place between UE and network. We tried N3 and N78 bands with different bandwidth, SCS and srate settings but no success yet. Would you be able to share the config file parameters for UE simbox and open5GS?

Alternatively, we also tried a COTS UE Google Pixel 7 and were able to attach to 5G using srsRAN and open5GS but there is no data connectivity. We tried changing the srs config file parameters with different bandwidth and srate values but no success.

**Reply by Zplusless** [reply]
Hi @ismagom ， I'm confused. Shouldn't the top five digits of IMSI be the same as PLMN?
Do you mean set PLMN as 90170 in the config file of AMF. Set imsi as 00101xxxxx...   at sim card and open5gs webui?

**Reply by brendan-mcauliffe** [reply]
HI @Zplusless, Ismael is referring to a set-up that forces the UE to use roaming. In such a set-up the gNB and core are configured to use a PLMN such as 00101, while the UE/SIM is configured with PLMN 90170. When the UE is registered in the core it should use the IMSI as defined in the SIM (90170......). Hope that clears things up.

**Reply by Zplusless** [reply]
Thanks @brendan-mcauliffe ， Your reply cleared up my confusion

**Comment by ismagom** [comment]
> One question for the community and developers: is anyone else using an Amarisoft AMARI UE Simbox? With our current configuration the UE Simbox is experiencing >90% bit-error rate and only a few kbps on iperf3 while our other UEs easily get >6 Mbps on iperf3. If anyone else out there is using an Amarisoft AMARI UE Simbox with srsRAN, I'm hoping to compare set ups and throughput numbers.

Hi @jtplatt99, 

Thanks for your feedback. We use Amari UE Simbox on a daily basis in our CI/CD. We always get very low BLER on all BWs and configurations. Can you explain us more about your setup? Things like wiring, CPUs, USRPs, etc are very helpful. Also if you could  attach logs with `all_level: info`  and `phy_level: debug` would also be helpful?

Thanks

**Reply by jtplatt99** [reply]
Hi @ismagom ,

Thanks for the quick reply. Our system has an Intel Core i7-9850H CPU with 16 GB memory. We’re running both srsRAN and Open5GS on this machine in separate Docker containers. We’re using a USRP N320 connected over 10GbE fiber though an Aquantia Thunderbolt3 NIC. We’re using the internal reference clock on the N320. You can see the srsRAN config file we’re using in my initial post. From the N320 we’re feeding into antennas for OTA transmission to phones and the Amari UE Simbox. The Amari UE Simbox is being provided a GPS reference signal for synchronization. Let me know if there are any more details on my setup that I can provide.

I also wanted to note that we’ve also evaluated similar set ups at 10 MHz and 40 MHz channel bandwidths and haven’t had BER problems in these scenarios. We’re only having problems with the Amari UE Simbox at 20 MHz channel bandwidth specifically.

Here are some logs files based on what you requested: [logs.zip](https://github.com/srsran/srsRAN_Project/files/11052535/logs.zip). The files “10MHzAmari.log”, “20MHzOP9Pro.log”, and “40MHzAmari.log” are logs from scenarios that seem to be working as intended. The file “20MHzAmari.log” is from the scenario I described where there is extremely high downlink BER. If it isn’t clear, 10MHz, 20MHz, and 40MHz in the first part of the file name correspond to the channel bandwidth being used, while the second part of the name is the UE that is connected. To provide some load on the network, the UEs are running a downlink iperf3 UDP test.

Thanks

**Reply by ismagom** [reply]
Hi @jtplatt99 , could you try setting the `pci: 1` in the config file? 

Thanks

**Reply by ismagom** [reply]
And if it doesn't work, could you attach Amari UE logs please?

**Reply by jtplatt99** [reply]
Hi @ismagom , I tried setting `pci: 1` as requested. With this set the downlink BER started low, but quickly increased to the levels I'd been seeing on `pci: 0` after a few seconds. Here are the Amarisoft UE logs from both scenarios: [AmarisoftLogs.zip](https://github.com/srsran/srsRAN_Project/files/11065170/AmarisoftLogs.zip)

Could you provide any additional insight into why it might be expected for `pci: 1` to perform better than `pci: 0`? Thanks!

**Reply by ismagom** [reply]
There seems to be some kind of RF problem, some variations in the received power, see for instance the `epre` and `snr` here: 

```
10:33:09.321 [PHY] DL 0001 00 4601   126.5 PDSCH: harq=6 prb=0:51 symb=2:12 k1=4 CW0: tb_len=1313 mod=4 rv_idx=3 cr=0.48 retx=1 crc=OK snr=13.9 epre=-99.6
10:33:09.324 [PHY] DL 0001 00 4601  126.10 PDSCH: harq=0 prb=0:51 symb=2:12 k1=7 CW0: tb_len=1313 mod=4 rv_idx=0 cr=0.48 retx=0 crc=KO snr=13.3 epre=-100.0
10:33:09.324 [PHY] DL 0001 00 4601  126.11 PDSCH: harq=2 prb=5:3 symb=2:12 k1=6 CW0: tb_len=80 mod=4 rv_idx=0 cr=0.48 retx=0 crc=KO snr=9.8 epre=-104.5
10:33:09.325 [PHY] DL 0001 00 4601  126.12 PDSCH: harq=3 prb=3:19 symb=2:12 k1=6 CW0: tb_len=488 mod=4 rv_idx=2 cr=0.48 retx=0 crc=KO snr=9.0 epre=-103.5
10:33:09.325 [PHY] DL 0001 00 4601  126.13 PDSCH: harq=1 prb=5:3 symb=2:12 k1=5 CW0: tb_len=80 mod=4 rv_idx=3 cr=0.48 retx=0 crc=KO snr=9.8 epre=-105.2
10:33:09.326 [PHY] DL 0001 00 4601  126.14 PDSCH: harq=4 prb=0:51 symb=2:12 k1=5 CW0: tb_len=1313 mod=4 rv_idx=3 cr=0.48 retx=1 crc=OK snr=13.5 epre=-100.3
10:33:09.326 [PHY] DL 0001 00 4601  126.15 PDSCH: harq=5 prb=3:19 symb=2:12 k1=4 CW0: tb_len=488 mod=4 rv_idx=1 cr=0.48 retx=0 crc=KO snr=8.9 epre=-103.8
10:33:09.329 [PHY] DL 0001 00 4601   127.0 PDSCH: harq=6 prb=11:31 symb=2:12 k1=7 CW0: tb_len=800 mod=4 rv_idx=0 cr=0.48 retx=0 crc=KO snr=5.1 epre=-103.6
10:33:09.329 [PHY] DL 0001 00 4601   127.1 PDSCH: harq=7 prb=0:51 symb=2:12 k1=6 CW0: tb_len=1313 mod=4 rv_idx=0 cr=0.48 retx=0 crc=KO snr=12.6 epre=-100.8
10:33:09.330 [PHY] DL 0001 00 4601   127.3 PDSCH: harq=0 prb=0:51 symb=2:12 k1=5 CW0: tb_len=1313 mod=4 rv_idx=2 cr=0.48 retx=1 crc=OK snr=12.5 epre=-101.0
10:33:09.331 [PHY] DL 0001 00 4601   127.4 PDSCH: harq=1 prb=0:51 symb=2:12 k1=5 CW0: tb_len=1313 mod=4 rv_idx=1 cr=0.48 retx=0 crc=KO snr=12.3 epre=-101.0
10:33:09.331 [PHY] DL 0001 00 4601   127.5 PDSCH: harq=2 prb=0:51 symb=2:12 k1=4 CW0: tb_len=1313 mod=4 rv_idx=2 cr=0.48 retx=0 crc=KO snr=12.1 epre=-101.1
10:33:09.334 [PHY] DL 0001 00 4601  127.10 PDSCH: harq=3 prb=0:18 symb=2:12 k1=7 CW0: tb_len=469 mod=4 rv_idx=3 cr=0.48 retx=0 crc=KO snr=9.2 epre=-105.1
10:33:09.334 [PHY] DL 0001 00 4601  127.11 PDSCH: harq=5 prb=3:19 symb=2:12 k1=6 CW0: tb_len=488 mod=4 rv_idx=0 cr=0.48 retx=1 crc=KO snr=6.6 epre=-105.8
10:33:09.335 [PHY] DL 0001 00 4601  127.12 PDSCH: harq=4 prb=5:3 symb=2:12 k1=6 CW0: tb_len=80 mod=4 rv_idx=0 cr=0.48 retx=0 crc=KO snr=6.7 epre=-107.8
10:33:09.335 [PHY] DL 0001 00 4601  127.13 PDSCH: harq=6 prb=3:19 symb=2:12 k1=5 CW0: tb_len=488 mod=4 rv_idx=2 cr=0.48 retx=0 crc=KO snr=6.4 epre=-105.8
10:33:09.336 [PHY] DL 0001 00 4601  127.14 PDSCH: harq=1 prb=0:51 symb=2:12 k1=5 CW0: tb_len=1313 mod=4 rv_idx=0 cr=0.48 retx=1 crc=OK snr=11.5 epre=-102.0
10:33:09.336 [PHY] DL 0001 00 4601  127.15 PDSCH: harq=2 prb=0:51 symb=2:12 k1=4 CW0: tb_len=1313 mod=4 rv_idx=3 cr=0.48 retx=1 crc=OK snr=11.5 epre=-102.1
10:33:09.339 [PHY] DL 0001 00 4601   128.0 PDSCH: harq=0 prb=11:31 symb=2:12 k1=7 CW0: tb_len=800 mod=4 rv_idx=0 cr=0.48 retx=0 crc=KO snr=6.2 epre=-103.9
10:33:09.340 [PHY] DL 0001 00 4601   128.3 PDSCH: harq=3 prb=3:19 symb=2:12 k1=5 CW0: tb_len=488 mod=4 rv_idx=1 cr=0.48 retx=0 crc=KO snr=5.9 epre=-107.0
10:33:09.341 [PHY] DL 0001 00 4601   128.4 PDSCH: harq=4 prb=0:51 symb=2:12 k1=5 CW0: tb_len=1313 mod=4 rv_idx=2 cr=0.48 retx=0 crc=KO snr=10.7 epre=-102.7
10:33:09.341 [PHY] DL 0001 00 4601   128.5 PDSCH: harq=6 prb=0:18 symb=2:12 k1=4 CW0: tb_len=469 mod=4 rv_idx=3 cr=0.48 retx=0 crc=KO snr=6.2 epre=-107.4
10:33:09.344 [PHY] DL 0001 00 4601  128.10 PDSCH: harq=7 prb=0:51 symb=2:12 k1=7 CW0: tb_len=1313 mod=4 rv_idx=2 cr=0.48 retx=1 crc=OK snr=10.5 epre=-103.2
10:33:09.344 [PHY] DL 0001 00 4601  128.11 PDSCH: harq=1 prb=5:3 symb=2:12 k1=6 CW0: tb_len=111 mod=6 rv_idx=0 cr=0.46 retx=0 crc=KO snr=3.1 epre=-110.6
10:33:09.346 [PHY] DL 0001 00 4601  128.14 PDSCH: harq=0 prb=0:18 symb=2:12 k1=5 CW0: tb_len=469 mod=4 rv_idx=2 cr=0.48 retx=0 crc=KO snr=5.4 epre=-107.8
10:33:09.346 [PHY] DL 0001 00 4601  128.15 PDSCH: harq=3 prb=0:18 symb=2:12 k1=4 CW0: tb_len=469 mod=4 rv_idx=0 cr=0.48 retx=0 crc=KO snr=7.2 epre=-107.0
10:33:09.349 [PHY] DL 0001 00 4601   129.0 PDSCH: harq=2 prb=11:31 symb=2:12 k1=7 CW0: tb_len=1153 mod=6 rv_idx=0 cr=0.46 retx=0 crc=KO snr=7.3 epre=-104.8
10:33:09.349 [PHY] DL 0001 00 4601   129.1 PDSCH: harq=4 prb=0:51 symb=2:12 k1=6 CW0: tb_len=1313 mod=4 rv_idx=3 cr=0.48 retx=1 crc=OK snr=9.2 epre=-104.1
10:33:09.350 [PHY] DL 0001 00 4601   129.2 PDSCH: harq=6 prb=3:19 symb=2:12 k1=6 CW0: tb_len=488 mod=4 rv_idx=1 cr=0.48 retx=0 crc=KO snr=4.8 epre=-108.2
10:33:09.350 [PHY] DL 0001 00 4601   129.3 PDSCH: harq=1 prb=0:51 symb=2:12 k1=5 CW0: tb_len=1889 mod=6 rv_idx=2 cr=0.46 retx=0 crc=KO snr=9.6 epre=-104.1
10:33:09.351 [PHY] DL 0001 00 4601   129.4 PDSCH: harq=5 prb=0:51 symb=2:12 k1=5 CW0: tb_len=1889 mod=6 rv_idx=0 cr=0.46 retx=0 crc=KO snr=9.4 epre=-104.2
```

Maybe you could try: 
- changing sampling rate or master clock rate in USRP
- changing DL ARFCN
- use wires to avoid interference
- in amari ue.conf you can try `phy.cell meas=1` in `log_options`. This will show the `freq_offset` measured by the UE. You want to see it very stable. 
- you can also try `phy.signal=1` also in `log_options` then use the amarisoft web gui and have a look at the constellation of those PDSCH packets


The test for the `pci: 0` was just an hypothesis. We've seen high retx when using cells other than `pci: 1` in some UEs, but Amarisoft is not the case.

**Reply by jtplatt99** [reply]
It seems like using a hardwired connection addressed this issue. I'm not sure what would be causing interference that specifically affects the Amari UE Simbox at 20 MHz bandwidth, but I will update this thread if I track down a configuration that addresses this issue with an OTA connection. Thanks!

**Comment by ChengDaHaI** [comment]
OnePlus Nord (AC2003) & Sierra EM9191 & Google Pixel 6 works under band n78@40MHz with USRP X310 (UHD 4.3.0)

**Reply by asheeshtripathi** [reply]
Hi @ChengDaHaI, what is the srate you set for 40 MHz in band 78? Also, is there any specific PLMN you configured? I am also using n78 on x310. Oneplus 8T and Telit 5G modem is able to scan the network but attach does not work. I have configured 10 MHz.

**Reply by ChengDaHaI** [reply]
` what is the srate you set for 40 MHz in band 78?`

46.08

`is there any specific PLMN you configured?`
001/01

**Reply by asheeshtripathi** [reply]
Thanks

**Comment by alvaroalfaro612** [comment]
I got Xiaomi **Redmi Note 10 5G** connected to different bands (28, 77, 78...). The main problem with this device is you need to activate 5G SA using the hidden option that is enabled by dialing :  *#*#726633#*#*

Also I have tried with new device **OnePlus Nord 2 CE Lite**, but in this case I am not able to find the network. I don´t know if a similar restriction as the xiaomi hidden menu is the problem in this case...

**Reply by Gotsche43** [reply]
Hi,
I have problem connecting my Redmi 10 5G. With the knowlegde that there can be many problems leading to the connectivity problems, I would just hear what PLMN did u use, and maybe the configuration file for the gNB?
Regards Magnus

**Comment by ChiefMeta** [comment]
I have samsung s22+. It does not see the network. Does anyone have the same experience?

**Comment by fllay** [comment]
I have ZTE MU5002/MU5001. It can be set to 5G SA mode from the menu in its touch screen and successfully  connected to the network. https://ztedevices.com/en-gl/mu5001/

Quectel  RM500Q-GL with Raspberry pi also successfully connected to the network. I have both of them connect at the same time for few hours.

**Reply by mhafidi24** [reply]
Hi @fllay,  I am using a Quectel RM520N-GL with Ubuntu 24.04.1 LTS, and I am trying to connect it to srsRAN gNB and Open5GS but it does not see my network. It only sees commercial networks from Telecom, o2 and Vodafone.

Could you please share some instructions on how did you achieve connecting your Quectel RM500Q-GL to the network?

PS: I successfully connected OnePlus Nord2 5G to my network

**Comment by mrouili** [comment]
Google pixel 7 Pro works fine with the srsRAN 5G SA mode and x310 USRP (SISO), band 78 TDD, 40-60Mhz.

**Reply by brendan-mcauliffe** [reply]
Thanks for reporting! I've added this to our docs, it will be live in the next update.

**Comment by warlock1302** [comment]


**Reply by pgawlowicz** [reply]
please start here: https://docs.srsran.com/projects/project/en/latest/tutorials/source/amariUE/source/index.html

**Comment by improperaffo** [comment]
Dear all,
we just tested a batch of phones. All of them were tested on the test MCC 001 (different MNCs: 01, 001, 02, 03) on band n78.
We used [5G Switch - Force 5G Only](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&ved=2ahUKEwjn46ryqr-EAxU79gIHHdy8CsAQFnoECA8QAQ&url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.sladjan.sava.petg%26hl%3Den%26gl%3DUS&usg=AOvVaw3Ad9qAcKUbB8MwVw7E547F&opi=89978449) to force the phones to be in 5G SA mode.
The working phones are:

- Realme 9
- Xiaomi Redmi Note 12 5G
- Motorola Moto G54
- Nokia G42 5G
- Oppo A78 5G
- Samsung A15 5G
- Samsung A23 5G

We had one that could not connect: 
- ~Xiaomi Poco X5 Pro 5G~

Previously we tested working the following phones:
- Oneplus 8
- Oneplus Nord CE 2 lite 5G
- Oneplus Nord CE 3 5G
- Samsung S23
- Samsung A76
- Asus Zenfone 8

**Reply by brendan-mcauliffe** [reply]
Thanks for reporting @improperaffo, I'll get these added to the docs soon.

**Reply by bcgordon3** [reply]
I have an unlocked Samsung A15 5G and while it works fine on a 00101 LTE network, I have been unable to get it to stay connected to the 5G network. Pixel 6 and 6 Pro stay fine. A15 registers and then drops right away never to return. I can see in the log that the core believe everything is fine, but A15 does not for some reason.

**Reply by improperaffo** [reply]
@bcgordon3 are you forcing the A15 to be in 5G only mode?

**Reply by bcgordon3** [reply]
@improperaffo I did have it forced to NR only. I also tried with the standard NR/LTE/WCDMA/GSM. Both had the same results. The phone registered with the core but did not stay attached at all. It was an immediate release by the phone. The core still thought it was attached. Any ideas?

**Reply by arman-maghsoudnia** [reply]
Hi, 
We tested the Galaxy A15 phone again. It works fine for us. Our idea is that the problem could be the OS software that you are using. I will post the information here, so you can compare them. 
Another problem that I think of is that the problem could be from the Sim card. Can you use the working sim card (like the one in the Pixel) and try using it with A15?

Phone info:
(https://github.com/srsran/srsRAN_Project/assets/93431904/aedae67a-9a77-44f4-9c36-af47ddb7be9f)

Software info:
(https://github.com/srsran/srsRAN_Project/assets/93431904/34ec7fb5-1ec0-40b1-a8a4-e43d6ea38a71)
(https://github.com/srsran/srsRAN_Project/assets/93431904/6ea25c83-51ae-43d4-86a2-b9a45a4332e0)

**Reply by bcgordon3** [reply]
Thanks for the tips. Looks like our model is slightly different and that may be the issue. I have model SM-156M/DSN. I also had an older firmware. I updated it to the latest, but that didn't fix the issue. Might be a version M vs B, which you have.

**Comment by Bourascom** [comment]
Dear all,
I’ve recently conducted tests on a batch of 5G phones. All devices underwent testing on the test PLMN 001-01 on Band28 , utilizing the ETTUS USRP B200 as the RF front-end without an external sync reference (relying on the internal clock). 
To ensure that the phones operated in 5G SA mode, I employed the 5G Switch - Force 5G Only feature.
The phones that successfully connected to the network are as follows:

- Google Pixel 7 Pro.
- Samsung Galaxy S23 Ultra (SM-S91B/DS).
- Samsung Galaxy S22 Ultra (SM-S908B/DS).
- Samsung Galaxy Z Flip 4 (SM-F721B).

However, I encountered issues with the following devices. Although they detected the network, they failed to register:

- Samsung Galaxy S22 Plus (SM-S906U).
- Sasmung Galaxy S21 (SM-G991B/DS).

I am expecting another batch of phones for further testing. I will promptly share the results once I complete the tests. I hope that this information proves valuable for the user-community and developers.

**Reply by brendan-mcauliffe** [reply]
Thanks @Bourascom these have been added to the docs and will be made public in the next release.

**Reply by andreaterni7** [reply]
Hi @Bourascom, can you please share the config file? Thank you in advance

**Reply by Bourascom** [reply]
Hi 
Here is gnb configurations :

amf:
  addr: 127.0.0.5                                               # The address or hostname of the AMF.
  bind_addr: 127.0.0.1                                            # A local IP that the gNB binds to for traffic from the AMF.

ru_sdr:
  device_driver: uhd                                              # The RF driver name.
  device_args: type=b200,num_recv_frames=64,num_send_frames=64    # Optionally pass arguments to the selected RF driver
  #clock: external
  #sync: external
  srate: 23.04                                                    # RF sample rate might need to be adjusted according to selected bandwidth.
  otw_format: sc12
  tx_gain: 80                                                     # Transmit gain of the RF might need to adjusted to the given situation.
  rx_gain: 40                                                     # Receive gain of the RF might need to adjusted to the given situation.

cell_cfg:
  dl_arfcn: 153620                                                # ARFCN of the downlink carrier (center frequency).
  band: 28                                                        # The NR band.
  channel_bandwidth_MHz: 10                                       # Bandwith in MHz. Number of PRBs will be automatically derived.
  common_scs: 15                                                  # Subcarrier spacing in kHz used for data.
  plmn: "00101"                                                   # PLMN broadcasted by the gNB.
  tac: 7                                                          # Tracking area code (needs to match the core configuration).
  pci: 1                                                          # Physical cell ID.

log:
  filename: /tmp/gnb.log                                          # Path of the log file.
  all_level: debug                                                # Logging level applied to all layers. 

pcap:
  mac_enable: true                                               # Set to true to enable MAC-layer PCAPs.
  mac_filename: /tmp/gnb_mac.pcap                                 # Path where the MAC PCAP is stored.
  ngap_enable: true                                              # Set to true to enable NGAP PCAPs.
  ngap_filename: /tmp/gnb_ngap.pcap                               # Path where the NGAP PCAP is stored.

You are free to modify the channel bandwidth to suit your host machine’s capabilities. It is crucial to be mindful of the antenna operating band. I’ve dedicated considerable time to testing various bands to ensure compatibility with the antenna, especially since I lack a VNA (Vector Network Analyzer) tool for precise antenna analysis. This attention to detail is vital, as the wrong band selection has previously caused numerous problems. Also, make sure that the AMF IP address in the amf.yaml file is consistent and correctly configured.

**Reply by victorlupas** [reply]
Just tested on Samsung Galaxy S24, it works!

**Comment by initChat** [comment]
I have CPE that is tested with USRP B210 and open5GS on band n78. It worked well.
- WLINK G230 Industrial 5G Router
- Huawei model N5368X
- ZLTX model X21Y

For using iperf, make sure to set mtu of the UE to `1400` and mtu of smf at `1450`

I also used Quectel RM510Q-GL connected to Raspberry PI as a UE. It is quite unstable, sometimes it can connect to the network and sometimes it cannot. The AT command of the Quectel is useful for checking network performance.
- `AT+QCSQ` returns RSRP, SINR, RSRQ
- `AT+QENG="servingcell"`, returns MCC, MNC, ARFCN, RSRP, RSRQ

However, `AT+QPING` to test ping is not worked. So I normally tested ping without using AT command.

**Comment by robertfalkenberg** [comment]
Didn't see it in the list yet, so I put it here as well:

Works fine for a long time already:

* Telit FN990A28 

```
at#getfw

HOST FIRMWARE  : A0R.000021
SLOT   STATUS     CARRIER  VERSION         TMCFG
1      Activated  Generic  M0R.000002      2071        
```

**Comment by jjose001** [comment]
Hello Everyone, 
we have tested [Samsung A52s 5G](https://cacombos.com/device/SM-A528B) with MCC 214 on band n78.
We used '5G Switch - Force 5G Only' to force the phones to be in 5G SA mode with root privileges. 
![imagen](https://github.com/user-attachments/assets/fd24dd21-d4a4-4adb-88c4-50e61f614de7)

**Reply by jjose001** [reply]
Motorola G55 also works well.

**Reply by jjose001** [reply]
Internet access also configured in both modes, with Docker and in daemon mode, successfully enabling NAT rules for proper connectivity in each case.

**Comment by mohit4444** [comment]
Hello Everyone,

Pixel 8 Pro successfully connects with  internet access!

Our setup included:
    One USRP X310 connected to the workstation via a 1G link
    Google Pixel 8 Pro with a sysmoISIM-SJA5-9FV SIM card

Here are the configuration files we used to get everything working.

Update: Even Quectel RMU500-EK works with the same setup.
Thanks!

[pixel 8 pro.zip](https://github.com/user-attachments/files/20298496/pixel.8.pro.zip)

**Comment by Zplusless** [comment]
OnePlus 12 connects to the internet (srsRAN 24.04+USRP-X310), but the performance is not satisfying.

here are the details https://github.com/srsran/srsRAN_Project/discussions/1221. 

I'm still trying to configure it.

**Comment by jaiglesias-optare** [comment]
I am using Quectel RM530N-GL modem, connected to an Ubuntu 24.04, with sysmocom sysmoISIM-SJA2, Open5GS, and USRP B210. 
Tests ok for different bands (n40, n78)...

**Reply by mohit4444** [reply]
Hi @jaiglesias-optare 
Is it possible to use the Quectel RM530N-GL as a network interface for the host machine it's connected to — particularly if that machine is separate from the one running the 5G core and srsRAN?

**Reply by jaiglesias-optare** [reply]
Yes, of course!. I am running srsRAN and Open5GS (also with influxdb, metrics server, grafana and prometheus) using docker containers in a linux PC (running Ubuntu 24.04), with USRP B210 connected by USB,  and the Quectel RM530N-GL is connected by USB to a different Ubuntu 22.04 laptop. The two machines are in the same room, but several meters apart from each other.

**Reply by mohit4444** [reply]
@jaiglesias-optare Interesting! Are you able to get internet access on the laptop via Quectel RM530N-GL?

**Reply by jaiglesias-optare** [reply]
Yes! After enabling forwarding and add the NAT route in the Open5GS host (the corresponding docker container), I can get internet access.
![image](https://github.com/user-attachments/assets/f7d19b43-a879-48ca-9b56-3729a07dc35d)
![image](https://github.com/user-attachments/assets/04365327-8d02-4230-8f64-05fcff087f85)

**Reply by mohit4444** [reply]
@jaiglesias-optare This is helpful. Thank you!

**Comment by alektoja** [comment]
Has anyone been able to connect the OnePlus 11? Preferably in a configuration with a Benetel RU.

**Reply by brendan-mcauliffe** [reply]
Hi @alektoja, the OnePlus 11 should work fine.
